server.py

```diff
@@ -0,0 +1,218 @@
+
+    def listen(self):
+        """Sets the socket listening for clients"""
+        sock = self._get_socket()
+        sock.listen(self.BACKLOG)  # wait for client connection
+
+        while True:
+            # Establish connection with client
+            conn, addr = sock.accept()
+            print("Got connection from", addr)
+
+            # Receive data from client
+            data = conn.recv(self.BUFFER_SIZE)
+
+            # Filter data for message or content
+            data = self.filter_data(data, conn)
+
+            if data is not None:
+                print("Server received", repr(data))
+                self._received_data = data
+                self.decision_tree()
+                conn.send(Conversation.DataReceived)
+
+            conn.close()
+
+    def filter_data(self, message, connection):
+        """Is the incoming data a message or data?"""
+
+        # Is this a message? If so perform action
+        if message in Conversation:
+            match message:
+                case Conversation.CanISendData:
+                    connection.send(Conversation.SendDataFormat)
+                case Conversation.SendDataFormat:
+                    pass  # Client Side Only
+                case Conversation.PublicKeyReceived:
+                    connection.send(Conversation.SendData)
+                case Conversation.SendData:
+                    pass  # Client Side Only
+                case Conversation.DataReceived:
+                    pass  # Client Side Only
+
+            return None
+
+        # Are these settings?
+        elif 16 <= int(message) < 128:
+            self.parse_settings(message)
+            if self._security == SecurityLevel.UnEncrypted:
+                connection.send(Conversation.SendData)
+            return None
+
+        # If not a message then this is data
+        else:
+            return message
+
+    def parse_settings(self, total):
+        """Parses settings from client for Security Level, Data Source, and Data Format"""
+        # I'm pretty sure there is a better bitwise method of doing this,
+        # but it works
+        # Edit: I tried the bitwise version and I don't think it is superior
+        # It is also less readable than this version.
+
+        # Security
+        if total >= SecurityLevel.Encrypted:
+            self.set_security(SecurityLevel.Encrypted)
+        elif total >= SecurityLevel.UnEncrypted:
+            self.set_security(SecurityLevel.UnEncrypted)
+
+        # Remove either Security Value
+        total = total % SecurityLevel.UnEncrypted
+
+        # Source Type
+        if total >= Source.TextFile:
+            self.set_source(Source.TextFile)
+        elif total >= Source.Dictionary:
+            self.set_source(Source.Dictionary)
+
+        # Remove either Source Type
+        total = total % Source.Dictionary
+
+        # Format
+        if total > 0:
+            self.set_pickling_format(Format(total))
+
+    def decision_tree(self):

```

Comment:
Can you remove this for now? has it is not related to the UI nor the server configuration options.

Another thing, nested if/else statements should only be used in very exceptional situations (there is actually a "never nester" movement that is actively trying to fight against this practice). It is an easy way of creating bugs

Reply:
Please explain - I've not heard of this movement?

Reply:
It started as a joke movement that you can find on Youtube or Google, but it has substance to it. It's about simplifying logic. If you want just do as you wish, I was trying to help

Reply:
Well, I'm sceptical, but I'll refactor into more functions. :-)